base/dict.jl

```diff
@@ -75,12 +75,30 @@ function showdict{K,V}(io::IO, t::Associative{K,V}; compact = false)
                 print(io, typeof(t))
             end
             print(io, '(')
+
+            # @Duplicate of code in show(IO, Pair)
+            #
+            # Previously called `show` to print each Pair for the brackets
+            # that it inserted if either element of the Pair was itself a Pair.
+            # However, it calls `show_default` if the actual types of either element
+            # does not match the type parameters of the Pair.
+            # e.g: Pair{Any,Any}(1, 2)
+            print_pair_with_brackets = (io::IO, p::Pair) -> begin
```

Comment:
@vtjnash You'd end up with a non-trivial number of brackets, though.

`Dict{Any,Any}((1)=>(2=>0x33),(2)=>(3=>0x4444),(3)=>(25.0))`
`Dict{Any,Any}(1=>(2=>0x33),2=>(3=>0x4444),3=>25.0)`

Then again, that might be easier to read...